lib/ApparelSorter.js

```diff
@@ -213,25 +203,67 @@ function findRegexMatch(patterns, iteration, size) {
     }
     const sizeObj = {
         regex: "No Match", 
-        index: parseInt(size.replace(/[^\d.-]/g, '')), 
+        index: parseSizing(size), 
         size: size,
         normalized: size,
         sizeVal: extractSizeVal(size, size)
-    }
+    };
     return sizeObj;
 }
 
-const parseSizeVal = (size) => {
-    let sizeInt = parseInt(size) || 0;
-    const hasLengthParam = size.includes('Short') || size.includes('Long');
+const parseSizing = (size) => {
+    let sizeInt;
+    const isValid = (size) => isNaN(size) == false;
+    try {
+        // First, attempt to parse the size directly.
+        sizeInt = getNumericQuantity(size);
+
+        if (isValid(sizeInt)) {
```

Comment:
Wouldn't we want to run this function before we attempt `getNumericQuantity`? Seems like some type checks upfront might be good, but I might be looking at this wrong

Reply:
I think that's a fair call out -- `getNumericQuantity` returns `NaN` if we get anything except a number back, so if anything else is sent through, it'll fall through. I could see the checking we do is to make sure it's a string, but I hesitate to do any other checking than that.

Reply:
Ok, that makes sense!